### DISCUSSION TITLE
Any examples on how to use this with Pytorch lightning?

### DISCUSSION THREAD

**Discussion by shengy90** [TorchJD/torchjd#158] [General]
:wave: Hi, are there any examples on how we might be able to use this in conjunction with Pytorch Lightning?

Or would this require us to overwrite the `backward` [method](https://github.com/Lightning-AI/pytorch-lightning/blob/70f188b95c8635d69ec9ef72a10464b42b7e2849/src/lightning/pytorch/core/module.py#L1085C1-L1103C1) in our LightningModule to use `mtl_backward(loss1, loss2)` over the default loss.backward()?

**Comment by ValerianRey** [comment]
Hi! Sorry for the last answer, last week has been crazy in terms of workload.

We didn't test torchjd with Pytorch Lightning yet. Also, I've personally never even used pytorch lightning, so I can't answer you yet.

We'll probably work on this soon, and I agree that adding an example with pytorch lighting integration would be really nice.

**Comment by ValerianRey** [comment]
https://github.com/TorchJD/torchjd/pull/160
